### DISCUSSION TITLE
Check if user is Main Admin?

### DISCUSSION THREAD

**Discussion by Vodhin** [e107inc/e107#5034] [General]
How do you check to see if a user (other than USERID #1) is a Main Admin? I can't find an instance or reference anywhere

The installation of a plugin I'm working on needs to automatically add some new user classes (done), detect those new classes' IDs (done), and then assign the user to one of those new user classes (done). I would like this last step to be available only to users who are a Main Admin, since this plugin separates Admin users to isolate data (company 1 cannot access company 2's data and so on) and I need to have only Main Admins be able to create those Companies.

**Comment by Jimmi08** [comment]
`getperms('0')`
- look at login.php, there is an exception for the main admin. I copy it from it. 
OR
`(defined('ADMINPERMS') || ADMINPERMS == '0') `

**Reply by Vodhin** [reply]
Thank you. I knew it was defined somewhere. Found it in class2 of course, and had to do a tiny bit of experimenting: For anyone else looking to use this, this the zero must be compared as a sting: `if(ADMINPERMS === '0'){}` not  `if(ADMINPERMS === 0){}`
